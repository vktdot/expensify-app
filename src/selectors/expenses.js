import moment from 'moment';

export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const createdAtMoment = moment(expense.createdAt)
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true
        const endDateMatch = endDate? endDate.isSameOrAfter(createdAtMoment, 'day') : true
        const textMatch = typeof text!== 'string' || expense.description.toLowerCase().includes(text.toLowerCase())  ;
        return startDateMatch && endDateMatch && textMatch;
    }).sort(function(a,b){
        if(sortBy === 'date'){
            return b.createdAt - a.createdAt
        }
        else{
            return b.amount - a.amount
        }
    })
}