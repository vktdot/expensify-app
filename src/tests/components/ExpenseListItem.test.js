import React from 'react'
import { shallow } from 'enzyme'
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses'

test('should generate expenses list item', () => {
    const wrapper = shallow(<ExpenseListItem expense={expenses[2]}/>)
    expect(wrapper).toMatchSnapshot();
})