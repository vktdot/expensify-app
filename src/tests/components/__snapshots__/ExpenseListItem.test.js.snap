// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate expenses list item 1`] = `
<Link
  className="list-item"
  replace={false}
  to="/edit/3"
>
  <div>
    <h3
      className="list-item__title"
    >
      Shoes
    </h3>
    <span
      className="list-item__sub-title"
    >
       
      December 27th, 1969
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    $250.00
  </h3>
</Link>
`;
